File: api/src/objects/doctors/doctor.go
package doctors

import (
	"github.com/satori/go.uuid"
)

var defaultImage = "https://nwhacksstorage.blob.core.windows.net/doctors/noimg.png"

// Doctor represents a doctor
type Doctor struct {
	ID            string   `json:"id"`
	Name          string   `json:"name"`
	Office        string   `json:"office"`
	OfficePhone   string   `json:"office_phone,omitempty"`
	PhotoURL      string   `json:"photo_url,omitempty"`
	Specializaion string   `json:"specialization"`
	Patients      []string `json:"patients"`
}

// NewDoctorConfig represents all the necessary info to create a new doctor
type NewDoctorConfig struct {
	Name           string `json:"name"`
	Office         string `json:"office"`
	OfficePhone    string `json:"office_phone,omitempty"`
	PhotoURL       string `json:"photo_url,omitempty"`
	Specialization string `json:"specialization"`
}

// NewDoctor is the constructor for the Doctor object
func NewDoctor(config NewDoctorConfig) *Doctor {
	d := &Doctor{
		ID:            uuid.Must(uuid.NewV4()).String(),
		Name:          config.Name,
		Office:        config.Office,
		OfficePhone:   config.OfficePhone,
		Specializaion: config.Specialization,
		PhotoURL:      defaultImage,
		Patients:      []string{},
	}
	if config.PhotoURL != "" {
		d.PhotoURL = config.PhotoURL
	}
	return d
}

File: iOS/beginnertableview/Models/Doctor.swift
//
//  Doctor.swift
//  NWHacks
//
//  Created by Felipe on 2019-01-26.
//  Copyright © 2019 Felipe. All rights reserved.
//

import Foundation

struct Doctor: Codable {
    var ID: String
    var Name: String
    var Office: String
    var OfficePhone: String?
    var Specializaion: String
    var PhotoURL: String?
    var Patients: [String]
    
    private enum CodingKeys: String, CodingKey {
        case ID = "id"
        case Name = "name"
        case Office = "office"
        case OfficePhone = "office_phone"
        case Specializaion = "specialization"
        case PhotoURL = "photo_url"
        case Patients = "patients"
    }
}

struct ListDoctors: Codable {
    var Doctors: [Doctor]
    
    private enum CodingKeys: String, CodingKey {
        case Doctors = "doctors"
    }
}
